File: back-end/netcore/VstepPractice.API/Models/Entities/SectionPart.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using VstepPractice.API.Common.Enums;

namespace VstepPractice.API.Models.Entities;

public class SectionPart : BaseEntity
{
    [Column("title")]
    [StringLength(255)]
    public string Title { get; set; } = string.Empty;

    [Column("instructions")]
    public string? Instructions { get; set; }

    [Column("content")]
    public string? Content { get; set; }

    [Column("orderNum")]
    public int OrderNum { get; set; }
    
    [Column("sectionType")]
    public SectionTypes SectionType { get; set; }
    [Column("type")]
    public SectionPartTypes Type { get; set; }

    [Column("examId")]
    public int ExamId { get; set; }

    [Column("parentId")]
    public int? ParentId { get; set; }

    // Navigation properties
    [ForeignKey(nameof(ExamId))]
    public virtual Exam Exam { get; set; } = default!;

    [ForeignKey(nameof(ParentId))]
    public virtual SectionPart? Parent { get; set; }

    public virtual ICollection<SectionPart> Children { get; set; } = new List<SectionPart>();
    public virtual ICollection<Question> Questions { get; set; } = new List<Question>();
}

File: back-end/db/models/section-part.js
'use strict'
const { Model } = require('sequelize')
module.exports = (sequelize, DataTypes) => {
  class SectionPart extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // define association here
      SectionPart.belongsTo(models.Exam, { foreignKey: 'examId', as: 'exam' })
      SectionPart.hasMany(models.SectionPart, {
        foreignKey: 'parentId',
        as: 'children',
      })

      SectionPart.belongsTo(models.SectionPart, {
        foreignKey: 'parentId',
        as: 'parent',
      })
      SectionPart.hasMany(models.Question, {
        foreignKey: 'passageId',
        as: 'questions',
      })
    }
  }

  SectionPart.init(
    {
      id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true,
      },
      title: DataTypes.STRING,
      instructions: { type: DataTypes.TEXT, allowNull: true },
      content: { type: DataTypes.TEXT, allowNull: true },
      orderNum: DataTypes.INTEGER,
      sectionType: DataTypes.INTEGER,
      type: DataTypes.INTEGER,
      examId: {
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
          model: 'Exams',
          key: 'id',
        },
        onDelete: 'CASCADE',
      },
      parentId: {
        type: DataTypes.INTEGER,
        allowNull: true,
        references: {
          model: 'SectionParts',
          key: 'id',
        },
        onDelete: 'CASCADE',
      },
    },
    {
      sequelize,
      modelName: 'SectionPart',
    },
  )
  return SectionPart
}
